--- a/data/themes/elarun/Main.qml
+++ b/data/themes/elarun/Main.qml
@@ -292,10 +292,23 @@
         }
     }
 
-    Component.onCompleted: {
+    function update_focus() {
         if (user_entry.text === "")
-            user_entry.focus = true
+            user_entry.forceActiveFocus();
         else
-            pw_entry.focus = true
+            pw_entry.forceActiveFocus();
+    }
+
+    //focus works in qmlscene
+    //but this seems to be needed when loaded from SDDM
+    //I don't understand why, but we have seen this before in the old lock screen
+    Timer {
+        interval: 200
+        running: true
+        onTriggered: update_focus()
+    }
+
+    Component.onCompleted: {
+        update_focus();
     }
 }
